server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -321,14 +321,9 @@ private static Fields generateTermVectorsFromDoc(IndexShard indexShard, TermVect
 
     private static ParsedDocument parseDocument(IndexShard indexShard, String index, BytesReference doc,
                                                 XContentType xContentType, String routing) {
-        MapperService mapperService = indexShard.mapperService();
-        DocumentMapperForType docMapper = mapperService.documentMapperWithAutoCreate();
-        ParsedDocument parsedDocument = docMapper.getDocumentMapper().parse(
-                new SourceToParse(index, "_id_for_tv_api", doc, xContentType, routing, Map.of()));
-        if (docMapper.getMapping() != null) {
-            parsedDocument.addDynamicMappingsUpdate(docMapper.getMapping());
-        }
```

Comment:
the dynamic mapping update here was super scary: we are generating term vectors for a provided artificial document, and it seems really wrong that that may cause a dynamic mapping update. Luckily, this had no effect as we were not doing anything with the dynamic mapping updates stored in this instance of ParsedDocument.